cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaRoller.java

```diff
@@ -346,8 +343,13 @@ private void restartIfNecessary(int podId, RestartContext restartContext)
             }
         } catch (ForceableProblem e) {
             if (restartContext.backOff.done() || e.forceNow) {
-                log.warn("{}: Pod {} will be force-rolled", reconciliation, podName(podId));
-                restartAndAwaitReadiness(pod, operationTimeoutMs, TimeUnit.MILLISECONDS);
+                if (canRoll(allClient, podId, 60_000, TimeUnit.MILLISECONDS)) {
+                    log.warn("{}: Pod {} will be force-rolled", reconciliation, podName(podId));
```

Comment:
Force-rolled suggests something went wrong. But in case of the controller this might be the first actual check of whether it can be restarted because before it was always deferred. So I think this message is right for other pods but not for the controller.